### PR TITLE
Feature/0296 possibility to update video streaming capabilities

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
Guidelines updates according to the proposal [SDL-0296](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md) https://github.com/smartdevicelink/sdl_core/issues/3465

This PR is **ready** for review.

### Summary
**Changelog**
Structs/index.md - add `AppCapability` struct, add `additionalVideoStreamingCapabilities` param
Enums/index.md - add `AppCapabilityType`

**Create** 
BC.OnAppCapabilityUpdated - index and assets

**Update**
BC.OnSystemCapabilityUpdated - update index.md; add OnSystemCapabilityUpdated_VIDEO_STREAMING diagram

### CODE REVIEW

File: docs/BasicCommunication/OnAppCapabilityUpdated/idex.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+## OnAppCapabilityUpdated
```
Comment (LitvinenkoIra) [inline docs/BasicCommunication/OnAppCapabilityUpdated/idex.md:1]: typo -> idex.md -> index.md

File: docs/BasicCommunication/OnAppCapabilityUpdated/idex.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Purpose
+: Inform that that a specific application capability has changed
+
+### Notification
+
+SDL sends OnAppCapabilityUpdated notification to HMI to provide supported by application <abbr title="Video Streaming Capabilities">VSC</abbr>
+
+!!! may
+
+HMI may switch video streaming resolution according to values provided by application in OnAppCapabilityUpdated notification. (_see also [OnSystemCapabilityUpdated](../OnSystemCapabilityUpdated/index.)_)
```
Comment (LitvinenkoIra) [inline docs/BasicCommunication/OnAppCapabilityUpdated/idex.md:18]: If HMI doesn't receive OnAppCapability updated - HMI should not send OnSysCapabilityUpdated and the user should not be able to change the resolution

File: docs/BasicCommunication/OnAppCapabilityUpdated/idex.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+!!!
+
+!!! note
+
+In case older application does not support sending `OnAppCapabilityUpdated` notification, HMI must treat this application as one that does not support dynamic resolution switching and allow only full-screen mode for it.
+
```
Comment (LitvinenkoIra) [inline docs/BasicCommunication/OnAppCapabilityUpdated/idex.md:24]: full-screen mode -> default screen mode

File: docs/BasicCommunication/OnAppCapabilityUpdated/idex.md (hunk 2 above)
Comment (LitvinenkoIra) [reply docs/BasicCommunication/OnAppCapabilityUpdated/idex.md:18]: of if HMI received OnAppCapabilityUpdated without additionalVideoStCaps inside

File: docs/BasicCommunication/OnSystemCapabilityUpdated/assets/OnSystemCapabilityUpdated_VIDEO_STREAMING.txt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+participant HMI
+
+==<align:center>Pre-conditions: 1.Streaming is in process\n2.App is subscribed to capabilities updates ==
+
+seq HMI notifies application of new Video Streaming Capabilities
+
+note over HMI:<align:center> User changes\nstreaming resolution
+
+HMI->HMI: Stop displaying video
+HMI->>SDL:<align:center> BC.OnSystemCapabilityUpdated\n(videoStreamingCapability=VideoStreamingCapability)
```
Comment (LitvinenkoIra) [inline docs/BasicCommunication/OnSystemCapabilityUpdated/assets/OnSystemCapabilityUpdated_VIDEO_STREAMING.txt:12]: need to add app_id as a parameter

File: docs/Common/Structs/index.md [hunk 5]
```diff
@@ -50,6 +50,7 @@
 |diagonalScreenSize|Float|false|minvalue: 0|The diagonal screen size in inches.|
 |pixelPerInch|Float|false|minvalue: 0|PPI is the diagonal resolution in pixels divided by the diagonal screen size in inches.|
 |scale|Float|false|minvalue: 1<br>maxvalue: 10|The scaling factor the app should use to change the size of the projecting view.|
+|additionalVideoStreamingCapabilities|Common.VideoStreamingCapability|false|array: true<br>minsize: 0<br>maxsize: 100|
```
Comment (LitvinenkoIra) [inline docs/Common/Structs/index.md:53]: minsize 1

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by LitvinenkoIra** [comment]
@jordynmackool @JackLivio This PR is ready for Livio review. Thank you!

File: docs/BasicCommunication/OnAppCapabilityUpdated/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+Purpose
+: Inform that that a specific application capability has changed
+
+### Notification
+
+SDL sends OnAppCapabilityUpdated notification to HMI to provide supported by application <abbr title="Video Streaming Capabilities">VSC</abbr>
+
+!!! may
+
+HMI may switch video streaming resolution according to values provided by application in OnAppCapabilityUpdated notification. (_see also [OnSystemCapabilityUpdated](../OnSystemCapabilityUpdated/index.)_)
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnAppCapabilityUpdated/index.md:18]:
```suggestion
HMI may switch video streaming resolution according to values provided by application in OnAppCapabilityUpdated notification. (_see also [OnSystemCapabilityUpdated](../OnSystemCapabilityUpdated)_)
```

File: docs/BasicCommunication/OnAppCapabilityUpdated/index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+### Sequence Diagrams
+
+|||
+Negotiation of Video Streaming Capabilities between HMI and Mobile Application
+![OnAppCapabilityUpdated](./assets/OnAppCapabilityUpdated.png)
+|||
+
+
+### JSON Message Examples
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnAppCapabilityUpdated/index.md:44]:
```suggestion
### JSON Message Examples
```

File: docs/BasicCommunication/OnAppCapabilityUpdated/index.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            "pixelPerInch":150,
+            "supportedFormats":[
+               {
+                  "protocol":"RAW",
+                  "codec":"H264"
+               }
+            ]
+         }
+      }
+   }
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnAppCapabilityUpdated/index.md:120]:
```suggestion
   }
}
```

File: docs/BasicCommunication/OnSystemCapabilityUpdated/index.md [hunk 9]
```diff
@@ -98,3 +114,25 @@ OnSystemCapabilityUpdated(APP_SERVICES, REMOVED)
   }
 }
 ```
+#### Videosteaming Switching Example Notification
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnSystemCapabilityUpdated/index.md:117]:
```suggestion

#### Videosteaming Switching Example Notification
```

File: docs/Common/Structs/index.md [hunk 10]
```diff
@@ -50,6 +50,7 @@
 |diagonalScreenSize|Float|false|minvalue: 0|The diagonal screen size in inches.|
 |pixelPerInch|Float|false|minvalue: 0|PPI is the diagonal resolution in pixels divided by the diagonal screen size in inches.|
 |scale|Float|false|minvalue: 1<br>maxvalue: 10|The scaling factor the app should use to change the size of the projecting view.|
+|additionalVideoStreamingCapabilities|Common.VideoStreamingCapability|false|array: true<br>minsize: 1<br>maxsize: 100|
```
Comment (ShobhitAd) [inline docs/Common/Structs/index.md:53]:
```suggestion
|additionalVideoStreamingCapabilities|Common.VideoStreamingCapability|false|array: true<br>minsize: 1<br>maxsize: 100||
```

**Review by ShobhitAd — CHANGES_REQUESTED** [review]
A new entry for `OnAppCapabilityUpdated` has to be added in config.json

**Comment by KhrystynaDubovyk** [comment]
> A new entry for OnAppCapabilityUpdated has to be added in config.json

done in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/223/commits/49651e176caf80bea2e77ecd5c6709e3f2ba45a3

File: docs/BasicCommunication/OnAppCapabilityUpdated/index.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+Sender
+: SDL
+
+Purpose
+: Inform that that a specific application capability has changed
+
+### Notification
+
+SDL sends OnAppCapabilityUpdated notification to HMI to provide supported by application <abbr title="Video Streaming Capabilities">VSC</abbr>
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnAppCapabilityUpdated/index.md:14]:
```suggestion
SDL sends OnAppCapabilityUpdated notification to HMI to provide <abbr title="Video Streaming Capabilities">VSC</abbr> supported by application
```

File: docs/BasicCommunication/OnAppCapabilityUpdated/assets/OnAppCapabilityUpdated.txt [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+SDL->HMI: UI.GetCapabilities
+HMI-->SDL:<align:center> UI.GetCapabilities\n(systemCapabilities=videoStreamingCapability,\n[additionalVideoStreamingCapabilities])
+end 
+
+note over App, HMI: RAI
+
+seq App receives VideoStreamingCapabilities supported by the system
+
+App->SDL:<align:center> GetSystemCapability\n(systemCapabilityType=VIDEO_STREAMING,\nsubscribe=true)
+SDL-->App:<align:center> GetSystemCapability\nsystemCapability=videoStreamingCapability,\n[additionalVideoStreamingCapabilities])
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnAppCapabilityUpdated/assets/OnAppCapabilityUpdated.txt:17]:
```suggestion
SDL-->App:<align:center> GetSystemCapability\n(systemCapability=videoStreamingCapability,\n[additionalVideoStreamingCapabilities])
```

File: docs/BasicCommunication/OnAppCapabilityUpdated/assets/OnAppCapabilityUpdated.txt [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+end
+
+seq App provides supported VideoStreamingCapabilities
+
+App->>SDL:<align:center> OnAppCapabilityUpdated\nappCapability=videoStreamingCapability,\n[additionalVideoStreamingCapabilities])
```
Comment (ShobhitAd) [inline docs/BasicCommunication/OnAppCapabilityUpdated/assets/OnAppCapabilityUpdated.txt:23]:
```suggestion
App->>SDL:<align:center> OnAppCapabilityUpdated\n(appCapability=videoStreamingCapability,\n[additionalVideoStreamingCapabilities])
```

**Review by ShobhitAd — CHANGES_REQUESTED** [review]

**Comment by ShobhitAd** [comment]
@KhrystynaDubovyk Thank you for making the changes. Could you also please re-generate the sequence diagram `docs/BasicCommunication/OnAppCapabilityUpdated/assets/OnAppCapabilityUpdated.png` based on the changes from https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/223/commits/570f6962d7aaf2a009dd5e6c547e0ab5ae491d43#diff-2db152cf1a1fa3403ab09f6a4b119c395190c5790055dffe0b20d8662e98a9ef

**Comment by KhrystynaDubovyk** [comment]
> @KhrystynaDubovyk Thank you for making the changes. Could you also please re-generate the sequence diagram `docs/BasicCommunication/OnAppCapabilityUpdated/assets/OnAppCapabilityUpdated.png` based on the changes from [570f696#diff-2db152cf1a1fa3403ab09f6a4b119c395190c5790055dffe0b20d8662e98a9ef](https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/commit/570f6962d7aaf2a009dd5e6c547e0ab5ae491d43#diff-2db152cf1a1fa3403ab09f6a4b119c395190c5790055dffe0b20d8662e98a9ef)

@ShobhitAd, fixed in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/223/commits/bea7440f1dd52df18c841c4addcbcacd4ecd4a47

**Review by ShobhitAd — APPROVED** [review]

**Comment by ShobhitAd** [comment]
@KhrystynaDubovyk please resolve merge conflicts on PR

**Review by ShobhitAd — APPROVED** [review]

**Pull request merged**